grim.go

```diff
@@ -189,19 +191,31 @@ func onHook(configRoot string, config *effectiveConfig, hook hookEvent, action h
 	writeHookEvent(resultPath, hook)
 
 	// TODO: do something with the err
-	notify(config, hook, "", GrimPending)
+	message, _ := notify(config, hook, "", GrimPending)
+	if logger != nil {
```

Comment:
Is a nil logger an acceptable use case?  Do you think it is ok ot have a nil logger?  If so, you should build an abstraction around it so that you dont have to do all these repeated nil checks.

If not, test it as the first thing in the function and return error immediately.